Homework/Khang_Nguyen/range-sum-query-2d-immutable.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @param {number[][]} matrix
+ */
+var NumMatrix = function(matrix) {
+    this.matrix = matrix
+
+    const colLen = matrix[0].length
+    const rowLen = matrix.length
+    const dp = []
+
+    for (let row = 0; row < rowLen; row++) {
+        dp.push([])
+        for (let col = 0; col < colLen; col++) {
```

Comment:
this should work but can try to use this trick to avoid dealing with 0 index: https://github.com/quangh33/Gen-5/pull/20/files#diff-fc8e4afa335b99bebd74679c19f03bcf60e53f7d6e729322bbc0d8354126b51d